### PR TITLE
updated datepicker documentation

### PR DESCRIPTION
**Pull request by xman343**
## Changes

Edited docs pages for the DatePicker component to bring it up-to-date with current implementation. Created new examples based on storybook.

## Testing

N/A

## Docs

N/A

### CODE REVIEW

File: apps/website/src/pages/docs/datepicker.mdx [hunk 1]
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
...
 When a user needs to schedule deadlines, timeframes, or for age verification purposes, we provide them with a date picker. It behaves much like a select menu. We currently have three different types of date pickers.
 
-## Appearance
+## Variants
 
-### Default
+### Basic
 
-The default date picker is a default menu button from which an interactive calendar expands on click. This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters.
+The basic date picker consists of a default menu button from which an interactive calendar expands on click.
```
Comment (gretanausedaite) [inline apps/website/src/pages/docs/datepicker.mdx:29]: The example does not have menu button from which a calendar expands. Maybe just describe that basic DatePicker consists of month skipping and date picking.
Then you could add a "Usage" section at the bottom of docs page with information that users can add DatePickers to menu buttons or simply on the page.

File: apps/website/src/pages/docs/datepicker.mdx [hunk 2]
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
...
+You are able to create a localized version of the date picker that uses local names for week days and months.
+
+<LiveExample src='DatePicker.localized.tsx' truncate={false}>
+  <AllExamples.DatePickerLocalizedExample client:load />
+</LiveExample>
+
+### Date + time selection
 
-There are two ways to select a date with the text field date picker; clicking on the button and selecting a date, or typing it out using the format shown in the help message underneath the field (DD-MM-YYYY). The date automatically fills the text field when directly selected in the calendar and is editable. It’s a more convenient way to offer editing in a context where the user might need to change the date often, or enter different dates for several items in a workflow.
+If you need to include the time, there are multiple ways to do so. Below is the default variant, which uses twenty-four hour / military time.
```
Comment (gretanausedaite) [inline apps/website/src/pages/docs/datepicker.mdx:47]: Remove "military-time", I think 24-hour format is enough.

File: apps/website/src/pages/docs/datepicker.mdx [hunk 3]
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
...
+## Variants
 
-### Default
+### Basic
 
-The default date picker is a default menu button from which an interactive calendar expands on click. This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters.
+The basic date picker consists of a default menu button from which an interactive calendar expands on click.
 
-### Text field
+<LiveExample src='DatePicker.basic.tsx' truncate={false}>
```
Comment (gretanausedaite) [inline apps/website/src/pages/docs/datepicker.mdx:31]: `truncate={false}` means that the code takes full space it needs, we usually do that only on main example. Other code examples should be truncated.

**Review by gretanausedaite — COMMENTED** [review]
You can remove `<Placeholder />` component. It's only temporary to tell people that docs are not finished.

Also, could you keep 1 component update per 1 PR? It's easier to track what's changed.

**Comment by gretanausedaite** [comment]
Could you merge this PR into `main` branch? We could release documentation update faster than major release.

File: apps/website/src/pages/docs/datepicker.mdx (hunk 1 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: Edited the description & added usage section.

File: apps/website/src/pages/docs/datepicker.mdx (hunk 3 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: Fixed.

File: apps/website/src/pages/docs/datepicker.mdx (hunk 2 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: Fixed.

**Comment by xman343** [comment]
> You can remove `<Placeholder />` component. It's only temporary to tell people that docs are not finished.
> 
> Also, could you keep 1 component update per 1 PR? It's easier to track what's changed.

Removed the placeholder component and reverted the commit for the unrelated component

File: apps/website/src/pages/docs/datepicker.mdx (hunk 1 above)
Comment (gretanausedaite) [reply apps/website/src/pages/docs/datepicker.mdx]: Add code example how to use DatePicker with menu button.

File: apps/website/src/pages/docs/datepicker.mdx [hunk 4]
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
 
 <p>{frontmatter.description}</p>
 
-<Placeholder componentPageName='core-datepicker' />
```
Comment (adamhock) [inline apps/website/src/pages/docs/datepicker.mdx:17]: Since this `Placeholder` was removed, the `Placeholder` import statement can also be removed

File: apps/website/src/pages/docs/datepicker.mdx [hunk 5]
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
...
+The basic date picker allows the user to pick a date from the calendar and skip a month at a time.
 
-### Text field
+<LiveExample src='DatePicker.basic.tsx' truncate={true}>
+  <AllExamples.DatePickerBasicExample client:load />
+</LiveExample>
 
-There are two ways to select a date with the text field date picker; clicking on the button and selecting a date, or typing it out using the format shown in the help message underneath the field (DD-MM-YYYY). The date automatically fills the text field when directly selected in the calendar and is editable. It’s a more convenient way to offer editing in a context where the user might need to change the date often, or enter different dates for several items in a workflow.
+This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters.
```
Comment (adamhock) [inline apps/website/src/pages/docs/datepicker.mdx:33]:
```suggestion
This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters in Table.
```
Also make sure to link `Table` to the Table documentation. There's an example of how to do this in the Expandable Block documentation page

File: apps/website/src/pages/docs/datepicker.mdx [hunk 6]
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
...
+  <AllExamples.DatePickerWithYearExample client:load />
+</LiveExample>
+
+## Usage
+
+A DatePicker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a DatePicker can be added to a menu button on your page so that the user can toggle its visibility.
+
 ## Props
 
 <PropsTable path={frontmatter.propsPath} />
```
Comment (adamhock) [inline apps/website/src/pages/docs/datepicker.mdx:87]: There should be a Related Component section (see Expandable Block)

File: apps/website/src/pages/docs/datepicker.mdx [hunk 7]
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
...
 
 To improve user experience, date pickers support year skipping buttons. Clicking on the double chevron icon will move the calendar a full year forward or back.
 
+<LiveExample src='DatePicker.withyear.tsx' truncate={true}>
+  <AllExamples.DatePickerWithYearExample client:load />
+</LiveExample>
+
+## Usage
+
+A DatePicker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a DatePicker can be added to a menu button on your page so that the user can toggle its visibility.
```
Comment (adamhock) [inline apps/website/src/pages/docs/datepicker.mdx:83]:
```suggestion
A date picker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a date picker can be added to a menu button on your page so that the user can toggle its visibility.
```

**Review by adamhock — COMMENTED** [review]
I would change the title of the section from `DatePicker` to `Date picker` (see Color picker)

These comments are just from reading the documentation. 

I was getting some build errors when building this branch, but they could be from my end. I will follow up on this tomorrow

File: apps/website/src/pages/docs/datepicker.mdx (hunk 7 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: done

File: apps/website/src/pages/docs/datepicker.mdx (hunk 6 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx:85]: added

File: apps/website/src/pages/docs/datepicker.mdx (hunk 5 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: done

File: apps/website/src/pages/docs/datepicker.mdx (hunk 4 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx:17]: removed

**Comment by xman343** [comment]
> I would change the title of the section from `DatePicker` to `Date picker` (see Color picker)
> 
> These comments are just from reading the documentation.
> 
> I was getting some build errors when building this branch, but they could be from my end. I will follow up on this tomorrow

fixed!

File: apps/website/src/examples/index.tsx [hunk 8]
```diff
@@ -242,9 +249,6 @@ export { default as ToggleSwitchInputGroupExample } from './ToggleSwitch.inputgr
 export { default as TooltipMainExample } from './Tooltip.main';
 export { default as TooltipPlacementExample } from './Tooltip.placement';
 
-export { default as TransferListMainExample } from './TransferList.main';
-export { default as TransferListWithLabelExample } from './TransferList.withlabel';
-
```
Comment (mayank99) [inline apps/website/src/examples/index.tsx:247]: revert this change. build is breaking because of it.

File: apps/website/src/pages/docs/datepicker.mdx (hunk 1 above)
Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: Added!

**Comment by xman343** [comment]
> 
I agree, will keep `dev` as the target

File: apps/website/src/pages/docs/datepicker.mdx (hunk 5 above)
Comment (adamhock) [reply apps/website/src/pages/docs/datepicker.mdx]: Make sure it links to the Table documentation like so

```mdx
[Table](table)
```

Comment (xman343) [reply apps/website/src/pages/docs/datepicker.mdx]: Fixed!

**Review by adamhock — APPROVED** [review]

**Review by gretanausedaite — APPROVED** [review]
